api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/dataGetter/SparqlQueryDataGetter.java

```diff
@@ -159,32 +200,70 @@ private Map<String, String> mergeParameters(
 	 * InitialBindings don't always work, and besides, RDFService doesn't accept
 	 * them. So do a text-based substitution.
 	 *
-	 * This assumes that every parameter is a URI. What if we want to substitute
-	 * a string value?
+	 * This assumes that every parameter is a URI unless data getter has specified
+	 * parameters. 
 	 */
-	private String bindParameters(String text, Map<String, String> merged) {
-		String bound = text;
-		for (String key : merged.keySet()) {
-			String value = merged.get(key);
-			if (value.startsWith("http://") || value.startsWith("https://")) {
-				/*
-				 * UQAM-Optimization if the "value" looks like an URI then wrap the value with the characters '<' '>'
-				 *
-				 */
-				bound = bound.replaceAll("([?$]" + key + ")([^a-zA-Z0-9_\\-])", "<" + value + ">$2");
-			} else {
-				bound = bound.replaceAll("([?$]" + key + ")([^a-zA-Z0-9_\\-])", value + "$2");
-			}
-		}
-		if (log.isDebugEnabled()) {
-			log.debug("parameters: " + merged);
-			log.debug("query before binding parameters:" + text);
-			log.debug("query after binding parameters: " + bound);
-		}
-		return bound;
-	}
+    private String bindParameters(String text, Map<String, String> parameters) {
+        ParameterizedSparqlString queryText = new ParameterizedSparqlString(text);
+        if (!isLegacyMode()) {
+            substitute(parameters, uriParams, queryText, 
+                    (pss, key, value) -> pss.setIri(key, value));
+            substitute(parameters, stringParams, queryText, 
+                    (pss, key, value) -> pss.setLiteral(key, value));
+            substitute(parameters, intParams, queryText,
+                    (pss, key, value) -> pss.setLiteral(key, Integer.parseInt(value)));
```

Comment:
in the case of an exception in parsing integer (if n3 file has a mistake), it will be handled by handleException in DataGetterDirective.java, right?

Reply:
Yes.